yt/geometry/coordinates/spherical_coordinates.py

```diff
@@ -401,3 +459,38 @@ def sanitize_width(self, axis, width, depth):
             yw = zmax - zmin
             width = [xw, yw]
         return width
+
+
+def _get1d_views(x1, x2, x3):
```

Comment:
note: i added this because the cython function that gets called expects 1d arrays. I'm sure there's a proper way to make the cython function handle arrays of arbitrary dimensions and avoid these functions...

Reply:
(i just havent figured it out yet)

Reply:
Well, it's probably possible by accepting `float *` and manually unrolling, but that's not my favorite.  It's also possible you could receive them with no dimensionality declaration, then inside the cython function use a view function and assign to a dimensionality-specified array.

Reply:
think i figured this out today, next push includes a refactor that resolves this.